src/cargo/core/workspace.rs

```diff
@@ -468,6 +498,15 @@ impl<'cfg> Workspace<'cfg> {
         Ok(None)
     }
 
+    /// After the root of a workspace has been located, sets the custom_metadata, if it exists.
+    pub fn load_workspace_metadata(&mut self) -> CargoResult<()> {
+        if let Some(workspace_config) = self.load_workspace_config()? {
+            self.custom_metadata = workspace_config.custom_metadata;
+        }
+
+        Ok(())
```

Comment:
Can this method be removed, and instead just use a single line in `new`, maybe something like:

```rust
ws.custom_metadata = ws.load_workspace_config()?.and_then(|cfg| cfg.custom_metadata);
```

Reply:
Done